cmake/FindLLVM.cmake

```diff
@@ -8,20 +8,41 @@
 #  LLVM_MODULE_LIBS - list of llvm libs for working with modules.
 #  LLVM_VERSION     - Installed version.
 
-set (LIBCLANG_KNOWN_LLVM_VERSIONS 8.0.0 8.0 8
-  7.0.1 7.0.0 7.0 7
-  6.0.1 6.0.0 6.0 6
-  5.0.2 5.0.1 5.0.0 5.0 5
-  4.0.1 4.0.0 4.0
-  3.9.1 3.9.0 3.9
-  3.8.1 3.8.0 3.8
-  3.7.1 3.7.0 3.7
-  3.6.2 3.6.1 3.6.0 3.6
-  3.5.2 3.5.1 3.5.0 3.5
-  3.4.2 3.4.1 3.4
-  3.3
-  3.2
-  3.1)
+set (LIBCLANG_KNOWN_LLVM_VERSIONS 15.0.0
```

Comment:
you should be able to use the `find_package( ... REQUIRED )` (in the main cmake file) to harvest the information about installed LLVM versions. This `cmake/FindLLVM` should no longer be required.
Backward compatibility up to LLVM 3.1 looks like something we don't have anyway in the rest of the code.

Which exact components are mapped into the old libraries we were using a decade ago... is certainly a good question.

Reply:
It is mainly used to find `llvm-config` and to get the proper recommend flags. The name is a bit misleading imho.

I remove older versions because they are useless here, I agree with you. Probably there is something to do in `FindLibClang.cmake` rather than on this. Feel free to reopen the issue if I am mistaken/ I have missed something.